cabal-install/Distribution/Solver/Compat/Prelude.hs

```diff
@@ -0,0 +1,19 @@
+-- to suppress WARNING in "Distribution.Compat.Prelude.Internal"
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
+
+-- | This module does two things:
+--
+-- * Acts as a compatiblity layer, like @base-compat@.
+--
+-- * Provides commonly used imports.
+--
+-- This module is a superset of "Distribution.Compat.Prelude" (which
+-- this module re-exports)
```

Comment:
Just a note: I've left this module in place (instead of just switching to Dist.Compat.Prelude directory) comment because figure it might grow a bit over time.